e2e/test-component/scenarios/embedding-sdk/dashboard-click-behavior.cy.spec.tsx

```diff
@@ -0,0 +1,161 @@
+import { InteractiveDashboard } from "@metabase/embedding-sdk-react";
+
+import { H } from "e2e/support";
+import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
+import {
+  POPOVER_ELEMENT,
+  cartesianChartCircle,
+  describeEE,
+  popover,
+} from "e2e/support/helpers";
+import {
+  mockAuthProviderAndJwtSignIn,
+  mountSdkContent,
+  signInAsAdminAndEnableEmbeddingSdk,
+} from "e2e/support/helpers/component-testing-sdk";
+import { getSdkRoot } from "e2e/support/helpers/e2e-embedding-sdk-helpers";
+
+const { ORDERS, ORDERS_ID } = SAMPLE_DATABASE;
+
+describeEE("scenarios > embedding-sdk > dashboard-click-behavior", () => {
+  beforeEach(() => {
+    signInAsAdminAndEnableEmbeddingSdk();
+
+    // Make the PRODUCT_ID column a URL column for click behavior tests, to avoid having to create a new model
+    cy.request("PUT", `/api/field/${ORDERS.PRODUCT_ID}`, {
+      semantic_type: "type/URL",
+    });
+
+    H.createDashboardWithQuestions({
+      dashboardName: "Orders in a dashboard",
+      questions: [
+        {
+          name: "Orders",
+          query: { "source-table": ORDERS_ID, limit: 5 },
+        },
+        {
+          name: "Line chart with click behavior",
+          display: "line",
+          query: {
+            "source-table": ORDERS_ID,
+            aggregation: [["count"]],
+            breakout: [
+              ["field", ORDERS.CREATED_AT, { "temporal-unit": "month" }],
+            ],
+            limit: 5,
+          },
+        },
+      ],
+      cards: [
+        {
+          size_x: 12,
+          col: 0,
+          visualization_settings: {
+            column_settings: {
+              '["name","SUBTOTAL"]': {
+                click_behavior: {
+                  type: "link",
+                  linkType: "url",
+                  linkTemplate: "https://metabase.com",
+                  linkTextTemplate: "Link Text Applied",
+                },
+              },
+            },
+          },
+        },
+        {
+          size_x: 12,
+          col: 12,
+          visualization_settings: {
+            click_behavior: {
+              type: "link",
+              linkType: "url",
+              linkTemplate: "https://metabase.com",
+            },
+          },
+        },
+      ],
+    }).then(({ dashboard }) => {
+      cy.wrap(dashboard.id).as("dashboardId");
+    });
+
+    cy.signOut();
+
+    mockAuthProviderAndJwtSignIn();
+
+    cy.intercept("GET", "/api/dashboard/*").as("getDashboard");
+    cy.intercept("POST", "/api/dashboard/*/dashcard/*/card/*/query").as(
+      "dashcardQuery",
+    );
+  });
+
+  it("should not trigger url click behaviors in the sdk (metabase#51099)", () => {
+    // Spies to intercept opening external links.
+    // See "clickLink" in frontend/src/metabase/lib/dom.js to see what we are intercepting.
```

Comment:
I was in `lib/dom.js`, but I'm not really sure what part of it this is referring to.

Reply:
Ah, this is referring to the clickLink function which creates an `<a href>` element and clicks on it to open a link:

https://github.com/metabase/metabase/blob/76ca2833e0ec33bfcbee307530bc41283559b880/frontend/src/metabase/lib/dom.js#L339

Reply:
cool thanks. Not sure why I missed the function name you referred to.